### PR TITLE
Serialization docs

### PR DESCRIPTION
**Pull request by mazameli**
I vetted this content by @sbelak and asked him some clarifying questions, but I'm realizing I still have some questions about the behavior of this feature when dealing with system settings:

- I know permissions, user, and group settings are not copied over. Are there any other admin settings that are _not_ copied? (E.g. are white-labeling settings copied?)
- Does the `skip` and `update` flag affect admin panel and data model settings? Or will my settings always get changed to match the settings contained in the dump?
- If I have public sharing turned on for a given dashboard, does that setting get carried over to the loaded dashboard?
- Are personal collections copied or ignored?

### CODE REVIEW

**Comment by mazameli** [comment]
This is ready for review @sbelak + @salsakran.

File: docs/enterprise-guide/serialization.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
+## Copying contents of one Metabase instance to another
+
+Metabase's serialization feature allows you to create a snapshot, called a dump, of the contents of a Metabase instance that can then be loaded into another instance.
```
Comment (sbelak) [inline docs/enterprise-guide/serialization.md]: If someone has a more poetic name than dump, I'm happy to change the (admittedly not very thought-out) terminology.

File: docs/enterprise-guide/serialization.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+This lets you do things like create a set of dashboards in one Metabase instance and then easily copy those dashboards to a number of other Metabase instances that you've set up for your customers. You could also use this feature to enable a staging-to-production workflow for important dashboards or reports by dumping from a staging instance of Metabase and then loading that dump into your production instance(s).
```
Comment (sbelak) [inline docs/enterprise-guide/serialization.md]: Might be good to clarify whether this is also meant for backups or should backups be done by backing up the app DB.

Also, should we mention that the dumps are human-readable (ish) YAMLs which you can put under source control and audit?

File: docs/enterprise-guide/serialization.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+If your instance is currently running, you will need to stop it first before creating or loading a dump, unless your Metabase application database supports concurrent reads. The default application database type, H2, does not.
+
+### Creating a data dump
+
+To create a dump of a Metabase instance, use the following command in your terminal:
+
+`java -jar metabase.jar dump [dump_name] --user [example@example.com]`
+
+The optional `--user` flag is used to specify a default administrator account for cases when this dump is loaded into a blank Metabase instance. This user will also be marked as the creator of all artifacts that are copied over to the instance. This user's personal collection and its contents will also be included in the data dump. If this flag isn't specified, it's assumed that the instance into which you're loading already has an admin user.
```
Comment (sbelak) [inline docs/enterprise-guide/serialization.md]: ... if there isn't one, load will fail.

File: docs/enterprise-guide/serialization.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+### Loading a dump
+
+To load a dump into another instance of Metabase, use the following command, where `[my_dump]` is the path to the dump you want to load:
+
+`java -jar metabase.jar load [my_dump] --mode [skip/update] --on-error [continue/abort]`
+
+The `--mode` flag lets you specify what to do when encountering a duplicate dashboard, question, or any Admin Panel settings that already set (again, except for permissions and user settings, which are not currently included in data dumps). It can either `skip` that item and do nothing to it, or `update` it with the version being loaded. The default is `skip`.
+
+The `--on-error` flag allows you to specify whether the load process should keep going or stop when there's an error. The default is `continue`.
```
Comment (sbelak) [inline docs/enterprise-guide/serialization.md]: Even with abort we don't do a full rollback. Artefacts that were successfully loaded before the error will remain.

File: docs/enterprise-guide/serialization.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+Both of these flags are optional.
+
```
Comment (sbelak) [inline docs/enterprise-guide/serialization.md]: Maybe mention that (currently) the same version of MB has to be used on both ends?

**Comment by mazameli** [comment]
@camsaul These docs should be good to go for our next follow-on release once we have a branch for that going.

**Pull request merged**